src/script/descriptor.cpp

```diff
@@ -212,6 +215,18 @@ class OriginPubkeyProvider final : public PubkeyProvider
         ret = "[" + OriginString() + "]" + std::move(sub);
         return true;
     }
+    bool ToNormalizedString(const SigningProvider& arg, std::string& ret, bool priv) const override
+    {
+        std::string sub;
+        if (!m_provider->ToNormalizedString(arg, sub, priv)) return false;
+        if (sub[0] == '[') {
```

Comment:
6f5d66d2abfe241df9f27bf2b1e62e6604918034 : I'm confused about what's going on here. I assume related to this comment. Maybe explain the "weirdness" in a comment, and/or add a comment to the relevant test vector.
> you will still have some weirdness to handle when the bip32 pubkey provider returns an origin pubkey provider, but this itself is already within an origin pubkey provider.

Reply:
Added a comment.